### DISCUSSION TITLE
Bumping @dhis2/ui without duplicates

### DISCUSSION THREAD

**Discussion by martinkrulltott** [dhis2/notes#304] [Gotchas]
Lately when bumping the `@dhis2/ui` dependency with `yarn add @dhis2/ui`, the `yarn.lock` file will be full of duplicates of each sub-package of ui. Usually these can be removed with `npx yarn-deduplicate yarn.lock`, but running it just once doesn't do the trick any more. As seen here,  https://github.com/dhis2/data-visualizer-app/pull/1869/commits/c2bb79348dfb46fc15ec686a1ea0c73ee65beec8#diff-51e4f558fae534656963876761c95b83b6ef5da5103c4adef6768219ed76c2deR3071 , only the `"@dhis2/ui@^6.10.5", "@dhis2/ui@^6.23.5":` part was fixed, all of the sub-packages were still duplicated.

To solve this, run the initial `npx yarn-deduplicate yarn.lock`, then `yarn`, then another `npx yarn-deduplicate yarn.lock`.

An easy way to spot this is if you have a diff like `+600 -10` in `yarn.lock` after the `ui` dep was bumped, rather than a diff like `+400 -400` after the double-deduplicate has been run (example: https://github.com/dhis2/data-visualizer-app/pull/1869/commits/363afce0dc018072219465b8f6964660fb6da251)

**Comment by ghost** [comment]
Also, if you want to diagnose whether this is the problem you're having, you can use `yarn why @dhis2/ui` to see whether yarn has installed multiple versions for you of `@dhis2/ui`. And you can of course do the same with other libs. The main libs I've found good to check are: `@dhis2/ui`, `@dhis2/app-runtime`, `@dhis2/cli-app-scripts`
